std/math.d

```diff
@@ -5884,14 +5885,285 @@ T nextafter(T)(const T x, const T y) @safe pure nothrow @nogc
 real fdim(real x, real y) @safe pure nothrow @nogc { return (x > y) ? x - y : +0.0; }
 
 /****************************************
- * Returns the larger of x and y.
+ * Returns the largest (most positive) element of `nums`.
+ *
+ * Elements are compared after implicit conversion to a common floating-point
+ * type. If `nums` is empty or any element `isNaN`, a `nan` value will be returned.
  */
-real fmax(real x, real y) @safe pure nothrow @nogc { return x > y ? x : y; }
+auto fmax(T ...)(T nums)
+    if (allSatisfy!(ApplyRight!(isImplicitlyConvertible, real), T))
+{
+    pragma(inline, true);
+
+    alias F = AutoFImumType!T;
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, F)(nums);
+    else
+        return fimumImpl!(">", F)(nums);
+}
+
+/// ditto
+auto fmax(F, T ...)(T nums)
+    if (isFloatingPoint!F && allSatisfy!(ApplyRight!(isImplicitlyConvertible, F), T))
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F)(const F[] nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F, size_t count)(auto ref const(F[count]) nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (count <= 4)
+        return fimumImpl!(">", count, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums[]);
+}
+
+///
+pure @safe nothrow @nogc unittest
+{
```

Comment:
nitpick: maybe you can add an additional, simpler unittest case that can be shown to the user at the documentation. This one is quite technical